app/uk/gov/hmrc/helptosavefrontend/controllers/UpdateEmailAddressController.scala

```diff
@@ -72,4 +85,84 @@ class UpdateEmailAddressController @Inject() (val sessionCacheConnector:  Sessio
       }
     }
   } (redirectOnLoginURL = FrontendAppConfig.checkEligibilityUrl)
+
+  def emailVerified(emailVerificationParams: String): Action[AnyContent] = authorisedForHtsWithInfo { implicit request ⇒ implicit htsContext ⇒
+    EmailVerificationParams.decode(emailVerificationParams) match {
+
+      case Failure(e) ⇒
+        logger.warn(s"Could not decode email verification parameters: ${e.getMessage}", e)
+        Ok(views.html.register.email_verify_error(BadContinueURL))
+
+      case Success(params) ⇒
+        val result: EitherT[Future, String, Option[NSIUserInfo]] = for {
+          session ← sessionCacheConnector.get
+          userInfo ← updateSession(session, params)
+        } yield userInfo
+
+        result.fold({
+          e ⇒
+            logger.warn(e)
+            InternalServerError
+        }, { maybeNSIUserInfo ⇒
+          maybeNSIUserInfo.fold{
+            // this means they were ineligible
+            Ok(views.html.core.not_eligible())
+          }{ updatedNSIUserInfo ⇒
+            Ok(views.html.register.confirm_details(updatedNSIUserInfo))
+          }
+        })
+    }
+  } (redirectOnLoginURL = FrontendAppConfig.checkEligibilityUrl)
```

Comment:
do we need to name this redirectonLoginURL each time ? i guess we can simply say 
`} (checkEligibilityUrl)`

Reply:
true - but it makes it clear what the parameter is, otherwise there's a random URL floating around. But it depends on personl taste I guess